state-synchronizer/src/coordinator.rs

```diff
@@ -1064,6 +1071,20 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
             }
         }
 
+        // verify end-of-epoch-li against local state
+        // verify end-of-epoch-li's version corresponds to end-of-chunk version before
+        // passing it to execution
+        let end_of_epoch_li = if let Some(li) = end_of_epoch_li {
+            self.local_state.trusted_epoch.verify(&li)?;
+            if li.ledger_info().version() == new_version {
+                Some(li)
+            } else {
+                None
+            }
+        } else {
+            None
+        };
+
```

Comment:
can we ad a comment such that we do not pass in an end of epoch li unless we have all the transactions for that epoch?

Reply:
I think there is some quirkiness on the optimistic send starting on line 1048/1055, I believe that should be moved /after/ this check....